pkg/dsl/definition/template.go

```diff
@@ -160,3 +229,82 @@ func (td *traitDef) Complete(ctx process.Context) error {
 	}
 	return nil
 }
+
+// Output fetch the trait cr and set result to context
+func (td *traitDef) Output(ctx process.Context, client client.Client, name string) Template {
+	_, assists := ctx.Output()
+	for _, assist := range assists {
+		if assist.Name() != td.name {
+			continue
+		}
+		traitRef, err := assist.Object(nil)
+		if err != nil {
+			return td
+		}
+		traitCr, err := getObj(client, traitRef, name)
+		if err != nil {
+			return td
+		}
+		td.output = traitCr
+		return td
+	}
+	return td
+}
+
+// HealthCheck address health check for trait
+func (td *traitDef) HealthCheck() error {
+	if td.health == "" {
+		return nil
+	}
+	bi := build.NewContext().NewInstance("", nil)
+	if err := bi.AddFile("-", td.health); err != nil {
+		return err
+	}
+	if td.output != nil {
+		bt, _ := json.Marshal(td.output)
+		if err := bi.AddFile("output", fmt.Sprintf("output: %s", string(bt))); err != nil {
+			return err
+		}
+	} else {
+		return errors.WithMessagef(fmt.Errorf("there is no trait output cr for health check"), "trait %s health check", td.name)
+	}
+	insts := cue.Build([]*build.Instance{bi})
+	for _, inst := range insts {
+		if err := inst.Value().Err(); err != nil {
+			return errors.WithMessagef(err, "trait %s check", td.name)
+		}
+		isHealthVal := inst.Lookup("isHealth")
```

Comment:
IIRC, this means the "health" template is a cue expression, right?

Reply:
Yes.  And the "health" check just is used for the output of CR, which could be any kind of CR because of powerful cue expression.